components/custom-network/token-configuration.tsx

```diff
@@ -55,29 +55,42 @@ export default function TokenConfiguration({
   }
 
   function handleNFTTokenChange(e) {
-    changedDataHandler("tokens", { label: "validated", value: false });
     changedDataHandler("tokens", { label: "nftToken", value: e.target.value });
   }
 
+  function handleNetworkTokenChange(token: Token) {
+    changedDataHandler("tokens", { label: "networkToken", value: token.address });
+  }
+
   function setDeployedAddress(address) {
-    changedDataHandler("tokens", { label: "validated", value: true });
     changedDataHandler("tokens", { label: "nftToken", value: address });
   }
 
-  async function validateAddress() {
+  async function validateNFTAddress() {
```

Comment:
You can define this function inside the useEffect since I don't think its used anywhere else.